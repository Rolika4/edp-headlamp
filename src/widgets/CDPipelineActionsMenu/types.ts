export interface CDPipelineActionsMenuProps {
    isDetailsPage?: boolean;
}
