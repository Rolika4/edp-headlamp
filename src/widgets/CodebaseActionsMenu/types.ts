export interface CodebaseActionsMenuProps {
    isDetailsPage?: boolean;
}
