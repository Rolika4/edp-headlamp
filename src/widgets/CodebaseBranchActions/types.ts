import { ACTION_MENU_TYPES } from '../../constants/actionMenuTypes';
import { EDPCodebaseKubeObjectInterface } from '../../k8s/EDPCodebase/types';
import { EDPCodebaseBranchKubeObjectInterface } from '../../k8s/EDPCodebaseBranch/types';
import { ValueOf } from '../../types/global';

export interface CodebaseBranchActionsProps {
  data: {
    branch: EDPCodebaseBranchKubeObjectInterface;
    defaultBranch: string;
    codebaseData: EDPCodebaseKubeObjectInterface;
  };
  backRoute?: string;
  variant?: ValueOf<typeof ACTION_MENU_TYPES>;
  anchorEl?: HTMLElement;
  handleCloseResourceActionListMenu?: () => void;
}
