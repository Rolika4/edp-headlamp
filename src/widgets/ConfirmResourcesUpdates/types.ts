import { CRUD_TYPES } from '../../constants/crudTypes';

export interface ResourceToUpdate {
    actionType: CRUD_TYPES;
    kind: string;
    name: string;
}

export interface ConfirmResourcesUpdatesDialogForwardedProps {
    deleteCallback: (...any) => Promise<void>;
    customText?: string;
    resourcesArray: ResourceToUpdate[];
}
