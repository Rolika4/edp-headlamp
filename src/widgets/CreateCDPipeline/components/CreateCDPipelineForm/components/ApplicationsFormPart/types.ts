import { FieldEventTarget, FormNameObject } from '../../../../../../types/forms';

export interface ApplicationsFormPartProps {
    names: {
        [key: string]: FormNameObject;
    };
    handleFormFieldChange(eventTarget: FieldEventTarget): void;
}
