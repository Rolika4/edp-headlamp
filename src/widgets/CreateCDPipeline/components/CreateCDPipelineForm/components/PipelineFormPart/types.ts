import { FieldEventTarget, FormNameObject } from '../../../../../../types/forms';

export interface PipelineInfoFormPartProps {
    names: {
        [key: string]: FormNameObject;
    };
    handleFormFieldChange(eventTarget: FieldEventTarget): void;
    onPipelineNameChange: (pipelineNameFieldValue: string) => void;
}
