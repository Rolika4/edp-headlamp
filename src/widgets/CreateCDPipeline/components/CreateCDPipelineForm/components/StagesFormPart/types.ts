import React from 'react';
import { EDPCDPipelineStageKubeObjectInterface } from '../../../../../../k8s/EDPCDPipelineStage/types';
import { DeepPartial } from '../../../../../../types/global';

export interface StagesFormPartProps {
    setCreateStageDialogOpen: React.Dispatch<React.SetStateAction<boolean>>;
    stages: DeepPartial<EDPCDPipelineStageKubeObjectInterface>;
    onStageDelete: (idx: number) => void;
}
