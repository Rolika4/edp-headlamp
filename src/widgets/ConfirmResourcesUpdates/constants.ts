export const CONFIRM_RESOURCES_UPDATES_DIALOG_NAME = 'CONFIRM_RESOURCES_UPDATES';
