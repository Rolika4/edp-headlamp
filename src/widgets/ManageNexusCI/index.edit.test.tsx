/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/react';
import React from 'react';
import { TestWrapper } from '../../../mocks/wrappers/default';
import { SYSTEM_QUICK_LINKS } from '../../k8s/QuickLink/constants';
import { NexusCISecretMock } from '../../k8s/Secret/mocks/nexus-ci-secret.mock';
import { SecretKubeObjectInterface } from '../../k8s/Secret/types';
import { FORM_MODES } from '../../types/forms';
import { ManageNexusCI } from './index';

test('renders ManageNexusCI Edit component', () => {
  render(
    <TestWrapper>
      <ManageNexusCI
        formData={{
          nexusSecret: NexusCISecretMock as unknown as SecretKubeObjectInterface,
          ownerReference: undefined,
          nexusQuickLink: {
            apiVersion: 'v1.edp.epam.com/v1',
            kind: 'QuickLink',
            metadata: {
              name: SYSTEM_QUICK_LINKS.NEXUS,
              namespace: 'test-namespace',
              creationTimestamp: '',
              uid: '',
            },
            spec: {
              type: SYSTEM_QUICK_LINKS.NEXUS,
              url: 'https://test-nexus.com',
              visible: true,
              icon: '',
            },
            status: '',
          },
          mode: FORM_MODES.EDIT,
          handleClosePanel: jest.fn(),
        }}
      />
    </TestWrapper>
  );

  const dialog = screen.getByTestId('form');
  expect(dialog).toMatchSnapshot();
});
