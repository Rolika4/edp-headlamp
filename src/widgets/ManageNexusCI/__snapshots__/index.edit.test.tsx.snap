// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ManageNexusCI Edit component 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  data-testid="form"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded Mui-expanded MuiAccordion-gutters css-3ia7zi-MuiPaper-root-MuiAccordion-root"
    >
      <div
        aria-expanded="true"
        class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
        role="button"
        style="cursor: default;"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content Mui-expanded MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-k6p83p-MuiTypography-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-7ldjcq-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                style="margin-right: 0.3125rem;"
              >
                <div>
                  <span />
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                ci-nexus
              </div>
            </div>
          </h6>
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiAccordion-region"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                            style="display: flex;"
                          >
                            <div
                              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                            >
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                              >
                                <span
                                  class="makeStyles-labelWrap-4"
                                >
                                  <span
                                    class="MuiTypography-root MuiTypography-body1 makeStyles-label-5 css-1ezega9-MuiTypography-root"
                                  >
                                    Quick Link URL
                                  </span>
                                  <span />
                                </span>
                              </div>
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                                style="display: flex; align-items: flex-end;"
                              >
                                <div
                                  class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                                >
                                  <div
                                    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                                  >
                                    <div
                                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1k0n0u9-MuiInputBase-root-MuiOutlinedInput-root"
                                    >
                                      <input
                                        aria-invalid="false"
                                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                                        id="mui-3"
                                        name="externalUrl"
                                        placeholder="Enter URL"
                                        type="text"
                                        value="https://test-nexus.com"
                                      />
                                      <fieldset
                                        aria-hidden="true"
                                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                                      >
                                        <legend
                                          class="css-ihdtdm"
                                        >
                                          <span
                                            class="notranslate"
                                          >
                                            â€‹
                                          </span>
                                        </legend>
                                      </fieldset>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                            style="display: flex;"
                          >
                            <div
                              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                            >
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                              >
                                <span
                                  class="makeStyles-labelWrap-4"
                                >
                                  <span
                                    class="MuiTypography-root MuiTypography-body1 makeStyles-label-5 css-1ezega9-MuiTypography-root"
                                  >
                                    URL
                                  </span>
                                  <span />
                                </span>
                              </div>
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                                style="display: flex; align-items: flex-end;"
                              >
                                <div
                                  class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                                >
                                  <div
                                    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                                  >
                                    <div
                                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-adornedEnd css-1wqgp0s-MuiInputBase-root-MuiOutlinedInput-root"
                                    >
                                      <input
                                        aria-invalid="false"
                                        class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                                        disabled=""
                                        id="mui-5"
                                        name="url"
                                        placeholder="Enter URL"
                                        type="text"
                                        value="https://test-url.com"
                                      />
                                      <div
                                        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                                      >
                                        <button
                                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                                          tabindex="0"
                                          type="button"
                                        >
                                          <span />
                                          <span
                                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                                          />
                                        </button>
                                      </div>
                                      <fieldset
                                        aria-hidden="true"
                                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                                      >
                                        <legend
                                          class="css-ihdtdm"
                                        >
                                          <span
                                            class="notranslate"
                                          >
                                            â€‹
                                          </span>
                                        </legend>
                                      </fieldset>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                            style="display: flex;"
                          >
                            <div
                              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                            >
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                              >
                                <span
                                  class="makeStyles-labelWrap-4"
                                >
                                  <span
                                    class="MuiTypography-root MuiTypography-body1 makeStyles-label-5 css-1ezega9-MuiTypography-root"
                                  >
                                    User
                                  </span>
                                  <span />
                                </span>
                              </div>
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                                style="display: flex; align-items: flex-end;"
                              >
                                <div
                                  class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                                >
                                  <div
                                    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                                  >
                                    <div
                                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1k0n0u9-MuiInputBase-root-MuiOutlinedInput-root"
                                    >
                                      <input
                                        aria-invalid="false"
                                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                                        id="mui-7"
                                        name="username"
                                        placeholder="Enter user name"
                                        type="text"
                                        value="test-username"
                                      />
                                      <fieldset
                                        aria-hidden="true"
                                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                                      >
                                        <legend
                                          class="css-ihdtdm"
                                        >
                                          <span
                                            class="notranslate"
                                          >
                                            â€‹
                                          </span>
                                        </legend>
                                      </fieldset>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                            style="display: flex;"
                          >
                            <div
                              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                            >
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                              >
                                <span
                                  class="makeStyles-labelWrap-4"
                                >
                                  <span
                                    class="MuiTypography-root MuiTypography-body1 makeStyles-label-5 css-1ezega9-MuiTypography-root"
                                  >
                                    Password
                                  </span>
                                  <span />
                                </span>
                              </div>
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                                style="display: flex; align-items: flex-end;"
                              >
                                <div
                                  class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                                >
                                  <div
                                    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                                  >
                                    <div
                                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1wqgp0s-MuiInputBase-root-MuiOutlinedInput-root"
                                    >
                                      <input
                                        aria-invalid="false"
                                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                                        id="mui-9"
                                        name="password"
                                        placeholder="Enter password"
                                        type="password"
                                        value="test-password"
                                      />
                                      <div
                                        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                                      >
                                        <button
                                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                                          tabindex="0"
                                          type="button"
                                        >
                                          <span />
                                          <span
                                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                                          />
                                        </button>
                                      </div>
                                      <fieldset
                                        aria-hidden="true"
                                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                                      >
                                        <legend
                                          class="css-ihdtdm"
                                        >
                                          <span
                                            class="notranslate"
                                          >
                                            â€‹
                                          </span>
                                        </legend>
                                      </fieldset>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1xu1mzw-MuiGrid-root"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                      >
                        <button
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="0"
                          type="button"
                        >
                          <span />
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                          >
                            <button
                              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1llbb2a-MuiButtonBase-root-MuiButton-root"
                              disabled=""
                              tabindex="-1"
                              type="button"
                            >
                              undo changes
                            </button>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                          >
                            <button
                              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-vw8gqz-MuiButtonBase-root-MuiButton-root"
                              disabled=""
                              tabindex="-1"
                              type="button"
                            >
                              save
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
